sudoku/src/domain/sudoku-model.ts

```diff
@@ -0,0 +1,98 @@
+const sudoku_example = [5, 3, 0, 0, 7, 0, 0, 0, 0, 6, 0, 0, 1, 9, 5, 0, 0, 0, 0, 9, 8, 0, 0, 0, 0, 6, 0, 8, 0, 0, 0, 6, 0, 0, 0, 3, 4, 0, 0, 8, 0, 3, 0, 0, 1, 7, 0, 0, 0, 2, 0, 0, 0, 6, 0, 6, 0, 0, 0, 0, 2, 8, 0, 0, 0, 0, 4, 1, 9, 0, 0, 5, 0, 0, 0, 0, 8, 0, 0, 7, 9];
+
+const SUDOKU_ROWS = 9;
+const SUDOKU_COLS = 9;
+const NUM_OF_SQUARES = SUDOKU_ROWS * SUDOKU_COLS;
+//SQUARES = list(range(NUM_OF_SQUARES))
+
+
+function printSudoku(sudoku: number[]): void {
+   for(let row=0; row<SUDOKU_ROWS ;row++){
+      let s = '';
+      let delimiter = '------+-------+------';
+      for(let col=0; col<SUDOKU_COLS ;col++){
+         if (sudoku_example[row*SUDOKU_ROWS+col] == 0) {
+            s +='  '
+         } else {
+            s += sudoku_example[row*SUDOKU_ROWS+col].toString()+' ';
+         }                  
+         if((col+1)%3 === 0 && (col+1)<SUDOKU_COLS){
+            s+='| ';
+         }
+      }
+      console.log(s);
+      if((row+1)%3 === 0 && (row+1)<SUDOKU_ROWS){
+            console.log(delimiter);
+         }  
+   }
+}
+
+function get_row_number(index: number): number {
+   return Math.floor(index/SUDOKU_ROWS);
+}
+
+function get_col_number(index: number): number {
+   return index % SUDOKU_COLS;
+}
+
+function get_box_number(index: number): number {   
+   let row_number = get_row_number(index);
+   let col_number = get_col_number(index);   
+   return (3*Math.floor(row_number/3) + Math.floor(col_number/3));   
+}
+
+function get_row(board: number[], row_number: number): number[] {
+   let row:number[] = new Array(SUDOKU_COLS);
```

Comment:
```suggestion
   const row:number[] = new Array(SUDOKU_COLS);
```
Du kan använda `const` här också då det alltid är en array oberoende om den är tom eller innehåller x antal värden. När vi har ESLint och Prettier konfigurerade så kommer IDE:n att varna när `let` används. 😊